### PR TITLE
Enforce user request about accessing the force sensor

### PR DESCRIPTION
**Pull request by alberth**
Make it impossible to access the force sensor if the user din't ask for it, even if the hardware supports it.

### CODE REVIEW

**Comment by MatthijsBurgh** [comment]
If we are going to enforce this, it should be made more generic for different kind of sensors.

**Comment by alberth** [comment]
It should become more generic indeed.

I considered that, but the pending issue is that there is currently no agreement on how to check in advance that a challenge will not fail on lack of hardware capabilities. For this reason, it seems unwise to spend time on generalizing code that may get deleted or moved to elsewhere. I started a discussion with a question about this in #767 .

This patch is just a fix wrt the semantics of `get_arm`. If you don't ask for the sensor there, you get an arm that does allow usage (if the hardware supports it). That is breaking the `get_arm` contract.

**Comment by LarsJanssenTUe** [comment]
@alberth do you want to update and merge this or do you want to take this into account in the new arm set-up you are making with Janno?

**Comment by alberth** [comment]
No idea what happened, rebase just worked, just needed a force-push.

Obviously, this change is enforcing getting access to arm functionality by first querying about it, which seems still undecided.
Besides that, just merge, when the arm is refactored, this sensor thing is a simple thing.

File: robot_skills/examples/example_move_down_until_force_sensor_edge_up.py [hunk 1]
```diff
@@ -4,6 +4,6 @@
 
 rospy.init_node('test_mode_down_until_force_sensor_edge_up', anonymous=True)
 robot = robot_constructor("hero")
-arm = robot.get_arm()
+arm = robot.get_arm(available_force_sensor=True)
```
Comment (LarsJanssenTUe) [inline robot_skills/examples/example_move_down_until_force_sensor_edge_up.py]: Make in line with master (force_sensor_required=True)

File: robot_skills/src/robot_skills/arms.py [hunk 2]
```diff
@@ -167,11 +175,14 @@ def send_gripper_goal(self, state, timeout=5.0, gripper_type=None, max_torque=0.
 
     @property
     def has_force_sensor(self):
+        # Check that the user enabled force sensor access.
+        self._test_die(self._allow_force_sensor, 'allow_force_sensor=' + str(self._allow_force_sensor),
+                       "Specify get_arm(..., available_force_sensor=True)")
```
Comment (LarsJanssenTUe) [inline robot_skills/src/robot_skills/arms.py]: available_force_sensor --> force_sensor_required

File: robot_skills/src/robot_skills/arms.py [hunk 3]
```diff
@@ -167,11 +175,14 @@ def send_gripper_goal(self, state, timeout=5.0, gripper_type=None, max_torque=0.
...
         return hasattr(self._arm, "force_sensor")
 
     def move_down_until_force_sensor_edge_up(self, timeout=10, retract_distance=0.01):
-        self._test_die(self.has_force_sensor, 'available_force_sensor=' + str(self.has_force_sensor),
-                       "Specify get_arm(..., available_force_sensor=True")
+        self._test_die(self.has_force_sensor, 'has_force_sensor=' + str(self.has_force_sensor),
+                       "Specify get_arm(..., available_force_sensor=True)")
```
Comment (LarsJanssenTUe) [inline robot_skills/src/robot_skills/arms.py]: available_force_sensor --> force_sensor_required

**Pull request merged**